src/LTL/Algorithm/TarjanModelChecker.cpp

```diff
@@ -102,22 +97,21 @@ namespace LTL {
 
         StateSet seen(_net, _k_bound);
         // master list of state information.
-        std::vector<centry_t> cstack;
+        light_deque<centry_t> cstack;
         // depth-first search stack, contains current search path.
-        std::stack<dentry_t<SuccGen>> dstack;
+        light_deque<dentry_t<SuccGen>> dstack;
 
-        _is_weak = successorGenerator.is_weak() && _shortcircuitweak;
```

Comment:
Again don't remove the successorGenerator call

Reply:
Similar idea as for NDFS, check lines 171-174 of TarjanModelChecker.cpp.